R/data_color.R

```diff
@@ -369,18 +371,18 @@ adjust_luminance <- function(colors,
   }
 
   # Get a matrix of values in the RGB color space
-  rgb_matrix <- t(grDevices::col2rgb(colors)) / 255
+  rgb_mat <- farver::decode_colour(colour = colors, alpha = TRUE)
 
-  # Obtain the alpha values
-  alpha <- get_alpha_vec(colors)
+  # Retain the alpha values
+  alpha <- rgb_mat[, "alpha"]
 
-  # Get a matrix of values in the Luv color space
-  luv_matrix <- grDevices::convertColor(rgb_matrix, "sRGB", "Luv")
+  # Get a matrix of values in the HCL color space
+  hcl_mat <- farver::convert_colour(colour = rgb_mat, from = "rgb", to = "hcl")
```

Comment:
Oh, is this the real reason you want to take the farver dependency?

Reply:
This is a major reason and also because farver reliability converts input hexadecimal-string colors and color names to a rgba color matrix

Reply:
Thanks for the conversation earlier. The farver dependency was removed entirely.